assets/aml-benchmark/components/src/inference_postprocessor/inference_postprocessor.py

```diff
@@ -90,129 +134,606 @@ def __post_init__(self) -> None:
-                result_df = pd.concat([result_df, probs_df], axis=1)
-        result_df.to_json(self.result, lines=True, orient='records')
+            if key != self.prediction_column_name:
+                row[self.prediction_column_name] = row.get(key)
+            predicted = row.get(self.prediction_column_name)
+            if isinstance(predicted, list):
+                try:
+                    out_string = jinja_template.render(predicted)
+                    pred_list.append(out_string)
+                except Exception as e:
+                    # "jinja2.exceptions.UndefinedError: 'list object' has no attribute 'split'"
+                    error_msg = "dictionary update sequence element"
+                    if isinstance(e, ValueError) and error_msg in e.args[0]:
+                        curr_pred_list = []
+                        for i in range(0, len(predicted)):
+                            curr_pred = {self.prediction_column_name: predicted[i]}
+                            out_string = jinja_template.render(curr_pred)
+                            curr_pred_list.append(out_string)
+                        pred_list.append(curr_pred_list)
+                    else:
+                        raise BenchmarkUserException._with_error(
+                            AzureMLError.create(BenchmarkUserError, error_details=e)
+                        )
+            else:
+                out_string = jinja_template.render(row)
+                pred_list.append(out_string)
+        if isinstance(pred_list[0], list) and len(pred_list[0]) > 1:
+            cols = [
+                f"{self.prediction_column_name}_{i+1}" for i in range(len(pred_list[0]))
+            ]
+        else:
+            cols = self.prediction_column_name
+        result_df[cols] = pred_list
+        # combine the records in one pandas dataframe and write it to the jsonl file.
+        result_df.to_json(self.result, lines=True, orient="records")
         return
 
-    def run(self) -> None:
+    def apply_generic_processor(self, out_string: str, row: dict = None) -> List:
+        """Processor steps."""
+        out_string = self.apply_remove_prompt_prefix(out_string, row)
+        out_string = self.apply_remove_prefixes(out_string)
+        out_string = self.apply_separator(out_string)
+        out_string = self.apply_find_first(out_string)
+        out_string = self.apply_extract_number(out_string)
+        out_string = self.apply_regex_expr(out_string)
+        out_string = self.apply_strip_characters(out_string)
+        out_string = self.apply_label_map(out_string)
+        return out_string
+
+    def extract_inferences(
+        self, result_df: pd.DataFrame, key: str = None, processor_order: List = None
+    ) -> pd.DataFrame:
+        """Extract inferences using generic method if no template or custom post-processor is provided."""
+        predicted_data = read_jsonl_files(resolve_io_path(self.prediction_dataset))

```

Comment:
This is small, but I would prefer all the file reading to be done in a separate method, and have a small method that just does the loop of extraction. It's a bit strange for a method to receive one dataset as  already loaded and the other to read on the fly.

Reply:
@alpatots I did not understand and need bit more elaboration on this. Let me know please when can we discuss this.

Reply:
@Nivi09 sorry about not being clear. It's a really small comment, but in general it seems strange (to me) to have a code like:

```python
def f(path_to_data, some_other_data):
    data = read_from(path_to_data)
    do_something(data, some_other_data)
```

I would expect to see 

```python
def f(some_data, some_other_data):
    do_something(data, some_other_data)
```